WordPress/WordPressApi/WordPressComOAuthClient.swift

```diff
@@ -0,0 +1,192 @@
+            }
+            success(authToken: authToken)
+
+            }, failure: { (task, error) in
+                if let httpURLResponse = task?.response as? NSHTTPURLResponse {
+                    let processedError = self.processError(error, response:httpURLResponse)
+                    failure(error: processedError)
+                    DDLogSwift.logError("Error receiving OAuth2 token: \(processedError)");
+                } else {
+                    failure(error: error)
+                    DDLogSwift.logError("Error receiving OAuth2 token: \(error)");
+                }
+            }
+        )
+    }
+
+    /** Requests a One Time Code, to be sent via SMS.
+     
+     - Parameters:
+        - username the account's username.
+        - password the account's password.
+        - success block to be called if authentication was successful.
+        - failure block to be called if authentication failed. The error object is passed as a parameter.
+    */
+    public func requestOneTimeCodeWithUsername(username: String, password:String,
+                                        success: () -> (), failure: (error: NSError) -> ())
+    {
+        let parameters = [
+            "username": username,
+            "password": password,
+            "grant_type": "password",
+            "client_id": clientID,
+            "client_secret": secret,
+            "wpcom_supports_2fa": true,
+            "wpcom_resend_otp": true
+        ]
+
+        sessionManager.POST("token", parameters:parameters, success:{ (task, responseObject) in
+            success()
+            }, failure: { (task, error) in
+                guard let httpURLResponse = task?.response as? NSHTTPURLResponse
+                else {
+                    failure(error: error)
+                    return
+                }
+                let processedError = self.processError(error, response:httpURLResponse)
+                // SORRY:
+                // SMS Requests will still return WordPressComOAuthErrorNeedsMultifactorCode. In which case,
+                // we should hit the success callback.
+                if processedError.code == WordPressComOAuthError.NeedsMultifactorCode.rawValue {
+                    success()
+                } else {
+                    failure(error: error)
+                }
+        })
+    }
+
+    private func processError(error: NSError, response:NSHTTPURLResponse) -> NSError {
+        guard response.statusCode >= 400 && response.statusCode < 500 && error.domain == AFURLResponseSerializationErrorDomain,
+        let responseData = error.userInfo[AFNetworkingOperationFailingURLResponseDataErrorKey] as? NSData,
+        let responseDictionary = (try? NSJSONSerialization.JSONObjectWithData(responseData, options:NSJSONReadingOptions())) as? [String:AnyObject],
+        let errorCode = responseDictionary["error"] as? String,
+        let errorDescription = responseDictionary["error_description"] as? String
+        else {
```

Comment:
As mentioned earlier, I think all this logic would fit better as a custom response serializer. Besides that, I'd be more specific with the status codes for known errors if we can: I think our OAuth2 endpoint will always return 400. Other errors should still be handled normally